### PR TITLE
[FIX] account: speedup _compute_tax_country_id

### PR DESCRIPTION
**Pull request by Aurelienvd**
Speedup computation of tax_country_id by first filtering on record with `fiscal_position_id.foreign_vat`. Then group records without foreign_vat by company_id and call `__setitem__` on each group. This reduces the number of calls to field.write and
provide an overall speedup on average. In the worst case, where each record has a fiscal_position_id with foreign_vat, the speed
is the same as the one prior to this PR.

#### speedup
Customer database with 60k account_moves, 150k account_move_lines, 1 company
Best case database with 1 company and all fiscal_position_id.foreign_vat is NULL
`_compute_tax_country_id` increasing size of recordset.

| Number of records | Before PR | After PR |
|:-------------------------:|:--------------:|:-----------:|
| 25 | 191ms | 25ms |
| 100 | 0.84s | 60ms |
| 500 | 4.03s | 220ms |
| 1000 | 7.37s | 456ms |

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2042,11 +2042,11 @@ def _compute_show_reset_to_draft_button(self):
 
     @api.depends('company_id.account_fiscal_country_id', 'fiscal_position_id.country_id', 'fiscal_position_id.foreign_vat')
     def _compute_tax_country_id(self):
-        for record in self:
-            if record.fiscal_position_id.foreign_vat:
-                record.tax_country_id = record.fiscal_position_id.country_id
-            else:
-                record.tax_country_id = record.company_id.account_fiscal_country_id
+        foreign_vat_records = self.filtered(lambda r: r.fiscal_position_id.foreign_vat)
+        for record in foreign_vat_records:
```
Comment (gawa-odoo) [inline addons/account/models/account_move.py:2046]: Is there a reason why you're not grouping here (with fiscal_position_id) if it improves that much if you group when you do with company_id?

Comment (gawa-odoo) [reply addons/account/models/account_move.py]: Yeah, most people would probably not have thousands of fiscal position, or often enough use the same

**Review by FlorianGilbert — COMMENTED** [review]
Hello
The code looks good to me but I just see a little typo in the commit, could you please fix it and it will be good to r+ :smile: 

> ... on record with firscal_position_id.foreign_vat. ....

change for

> ... on record with fiscal_position_id.foreign_vat. ....

Thank you!

**Comment by Aurelienvd** [comment]
@FlorianGilbert Well spotted 😮

**Review by FlorianGilbert — APPROVED** [review]
@robodoo r+

Thank you!

**Pull request closed without merging**